addOns/paramminer/src/main/java/org/zaproxy/addon/paramminer/UrlUtils.java

```diff
@@ -139,4 +142,95 @@ public static List<Map<String, String>> confirmUsableParameters(
         }
         return paramGroups;
     }
+
+    /**
+     * Removes all html tags from a given content string.
+     *
+     * @param str the content string to be stripped.
+     * @return a string without html tags.
+     */
+    public static String removeTags(String str) {
+        return str.replaceAll("(?s)<.*?>", "");
+    }
+
+    /**
+     * Returns a list of lines that are comon between two content Strings.
+     *
+     * @param body1 the first content string.
+     * @param body2 the second content string.
+     * @return a list of lines that are comon between two content Strings.
+     */
+    public static List<String> diffMap(String body1, String body2) {
+        List<String> diff = new ArrayList<>();
+        String[] body1Lines = StringUtils.split(body1, "\n");
+        String[] body2Lines = StringUtils.split(body2, "\n");
+        for (int i = 0; i < body1Lines.length; i++) {
+            if (body1Lines[i].equalsIgnoreCase(body2Lines[i])) {
+                diff.add(StringUtils.strip(body1Lines[i]));
+            }
+        }
+        return diff;
+    }
+
+    public static Factors defineAnomaly(
+            HttpMessage msg1, HttpMessage msg2, String param, String value, List<String> wordlist) {
+        Factors factors = new Factors();
+
+        String body1 = msg1.getResponseBody().toString();
+        String body2 = msg2.getResponseBody().toString();
+
+        if (msg1.getResponseHeader().getStatusCode() == msg2.getResponseHeader().getStatusCode()) {
+            factors.setSameCode(true);
+        }
+
+        if (msg1.getResponseHeader().getHeaders().equals(msg2.getResponseHeader().getHeaders())) {
+            factors.setHeaders(msg1.getResponseHeader().getHeaders());
+        }
+
+        if (body1.equalsIgnoreCase(body2)) {
+            factors.setSameBody(true);
+        } else if (StringUtils.countMatches(body1, '\n') == StringUtils.countMatches(body2, '\n')) {
+            factors.setLinesNumValue(StringUtils.countMatches(body1, '\n'));
+        } else if (removeTags(body1).equalsIgnoreCase(removeTags(body2))) {
+            factors.setPlainText(removeTags(body1));
+        } else if (!body1.isEmpty()
+                && !body2.isEmpty()
+                && (StringUtils.countMatches("\\n", body1)
+                        == StringUtils.countMatches("\\n", body2))) {
+            factors.setDiffMapLines(diffMap(body1, body2));
+        }
+
+        URI uri1 = msg1.getRequestHeader().getURI();
+        URI uri2 = msg2.getRequestHeader().getURI();
+        String path1 = "", path2 = "";
+        try {
+            path1 = uri1.getPath();
+        } catch (URIException e) {
+            logger.error("Invalid URL: " + uri1.toString(), e);
+        }
+        try {
+            path2 = uri2.getPath();
+        } catch (URIException e) {
+            logger.error("Invalid URL: " + uri2.toString(), e);
```

Comment:
This would suck but it isn't really preventing ZAP from running, I'd suggest warn or maybe even debug.

Reply:
Also maybe extract a method instead of doing the same thing twice.